libs/utils/executor.py

```diff
@@ -318,6 +318,30 @@ def _setup_sched_features(self, tc):
     def get_run_dir(target):
         return os.path.join(target.working_directory, TGT_RUN_DIR)
 
+    def _selinux_mode_unused(self):
+        return (self._old_selinux_mode is None or 'Disabled' in self._old_selinux_mode)
+
+    def _clear_selinux_mode(self):
+        try:
+            # First, save the old SELinux mode
+            self._old_selinux_mode = self.target.execute('getenforce')
```

Comment:
I'm not too hot on handling that through side effects, but I guess there's no better way to handle it now given it's all within Executor.

Reply:
Maybe, but this is the only place in the whole codebase where we use it - I could propose it as a devlib module instead, but that seems like a lot of work for a few lines. What do you think?

Reply:
I think it could easily fit as a couple of additional target APIs, not necessarily a new module.

Just after the 'adb_*' commands we have here:
   https://github.com/ARM-software/devlib/blob/master/devlib/target.py#L1361
perhaps it's fine to add a new context manager API which allows something like

```python
with selinux_disabled():
   # do your staff... and then SELinux settings will be updated if required.
```

But @valschneider is our context manager expert, is that possible?

Reply:
My endgame view here is to make as many features as possible available to devlib to have a cleaner/simpler interface for the "next generation" of workloads (and to be good open source citizens by pushing our devlib "addons" back to devlib).

I agree it's just a small change but IMHO it's worth the effort in the long run :)

In the end it's your call, we can have this merged and I'll come bother you when we finally get rid of `Executor` :)

Reply:
@derkling That ought to be possible, although I'd prefer a non context-manager version to be added first, and we can easily wrap it around one if needed.